pkg/agent/core/ngt/handler/grpc/upsert_test.go

```diff
@@ -17,103 +17,1557 @@ package grpc
 
 import (
 	"context"
+	"math"
 	"reflect"
 	"testing"
 
 	"github.com/vdaas/vald/apis/grpc/v1/payload"
 	"github.com/vdaas/vald/apis/grpc/v1/vald"
+	"github.com/vdaas/vald/internal/config"
+	"github.com/vdaas/vald/internal/conv"
+	"github.com/vdaas/vald/internal/core/algorithm/ngt"
 	"github.com/vdaas/vald/internal/errgroup"
 	"github.com/vdaas/vald/internal/errors"
+	"github.com/vdaas/vald/internal/net/grpc/codes"
+	"github.com/vdaas/vald/internal/net/grpc/status"
+	"github.com/vdaas/vald/internal/test/data/request"
+	"github.com/vdaas/vald/internal/test/data/vector"
 	"github.com/vdaas/vald/internal/test/goleak"
 	"github.com/vdaas/vald/pkg/agent/core/ngt/service"
 )
 
 func Test_server_Upsert(t *testing.T) {
 	t.Parallel()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
 	type args struct {
-		ctx context.Context
-		req *payload.Upsert_Request
-	}
-	type fields struct {
-		name              string
-		ip                string
-		ngt               service.NGT
-		eg                errgroup.Group
-		streamConcurrency int
+		ctx         context.Context
+		indexID     string
+		indexVector []float32
+		req         *payload.Upsert_Request
 	}
 	type want struct {
-		wantLoc *payload.Object_Location
-		err     error
+		code     codes.Code
+		wantUUID string
 	}
 	type test struct {
 		name       string
 		args       args
-		fields     fields
 		want       want
 		checkFunc  func(want, *payload.Object_Location, error) error
-		beforeFunc func(args)
+		beforeFunc func(args) (Server, error)
 		afterFunc  func(args)
 	}
-	defaultCheckFunc := func(w want, gotLoc *payload.Object_Location, err error) error {
-		if !errors.Is(err, w.err) {
-			return errors.Errorf("got_error: \"%#v\",\n\t\t\t\twant: \"%#v\"", err, w.err)
-		}
-		if !reflect.DeepEqual(gotLoc, w.wantLoc) {
-			return errors.Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", gotLoc, w.wantLoc)
+	defaultCheckFunc := func(w want, gotRes *payload.Object_Location, err error) error {
+		if err != nil {
+			st, ok := status.FromError(err)
+			if !ok {
+				errors.Errorf("got error cannot convert to Status: \"%#v\"", err)
+			}
+			if st.Code() != w.code {
+				return errors.Errorf("got code: \"%#v\",\n\t\t\t\twant code: \"%#v\"", st.Code(), w.code)
+			}
+		} else {
+			if uuid := gotRes.GetUuid(); w.wantUUID != uuid {
+				return errors.Errorf("got uuid: \"%#v\",\n\t\t\t\twant uuid: \"%#v\"", uuid, w.wantUUID)
+			}
 		}
 		return nil
 	}
+
+	const (
+		defaultInsertNum = 1000
+		dimension        = 128
+	)
+
+	utf8Str := "こんにちは"
+	eucjpStr, err := conv.Utf8ToEucjp(utf8Str)
+	if err != nil {
+		t.Error(err)
+	}
+	sjisStr, err := conv.Utf8ToSjis(utf8Str)
+	if err != nil {
+		t.Error(err)
+	}
+
+	defaultUpsertConfig := &payload.Upsert_Config{
+		SkipStrictExistCheck: true,
+	}
+	defaultInsertConfig := &payload.Insert_Config{
+		SkipStrictExistCheck: true,
+	}
+	beforeFunc := func(objectType string, insertNum int) func(args) (Server, error) {
```

Comment:
```suggestion
	defaultBeforeFunc := func(objectType string, insertNum int) func(args) (Server, error) {
```
It would be to format other definitions.